src/coord/src/coord.rs

```diff
@@ -4818,12 +4866,29 @@ pub mod fast_path_peek {
 
             // If we must build the view, ship the dataflow.
             let (peek_command, drop_dataflow) = match fast_path {
-                Plan::PeekExisting(id, key, map_filter_project) => (
-                    (id, key, timestamp, finishing.clone(), map_filter_project),
-                    None,
-                ),
+                Plan::PeekExisting(id, key, map_filter_project) => {
+                    let arrangement_instance = self
+                        .catalog
+                        .get_by_id(&id)
+                        .index()
+                        .expect("peaking from instance")
+                        .compute_instance;
+
+                    assert_eq!(
+                        compute_instance, arrangement_instance,
+                        "reading from an arrangement not on the target instance"
+                    );
```

Comment:
The compute layer will do this check, so it should be fine to elide here!

Reply:
You expect me to put a lot of faith that things happen in compute :D

Reply:
Haha, I can assure you @frankmcsherry is putting in these sorts of `assert!`s all over the place to prevent ADAPTER from holding COMPUTE wrong. :D